### ISSUE TITLE
No longer able to publish a NETStandard.Library without a runtime directive

### ISSUE THREAD

**Issue by jeffschwMSFT** [dotnet/cli#3119]
## Steps to reproduce
dotnet new
vi project.json
{
  "version": "1.0.0-*",
  "dependencies": {
    "NETStandard.Library" : "1.5.0-rc2-24027"
  },
  "frameworks": {
    "netstandard1.5": {
    }
  }
}
dotnet restore
dotnet build
dotnet publish

## Expected  behavior
Late RC2 builds would perform the publish and just the necessary dependencies would be published to the bin/<flavor>/netstandard1.5/publish directory

## Actual behavior
Can not find runtime target for framework '.NETStandard,Version=v1.5' compatible with one of the target runtimes: 'osx.10.11-x64'. Possible causes:
1. The project has not been restored or restore failed - run `dotnet restore`
2. The project does not list one of 'osx.10.11-x64' in the 'runtimes' section.

## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64

**Comment by davidfowl** [comment]
That's by design. If you want to go standalone you need to add a runtimes section and a specific RID.

**Comment by blackdwarf** [comment]
@davidfowl I think what @jeffschwMSFT wanted to do was to publish a library. This is not supported, as libraries are supposed to be `dotnet pack`-ed. We need guidance on this methinks.

**Comment by blackdwarf** [comment]
OK, will keep this issue and rename it to track work for a slightly better error message that will inform users about libraries and point them somewhere. 

@piotrpMSFT thoughts?

**Comment by brthor** [comment]
@blackdwarf Still going to rename this?

What do you think the error message should be?

Are we saying that anything but the `netcoreapp1.0` tfm cannot be published?

**Comment by blackdwarf** [comment]
@brthor I think the message should have something along the following:

```console
Can not find runtime target for framework '.NETStandard,Version=v1.5' compatible with one of the target runtimes: 'osx.10.11-x64'. Possible causes:
1. The project has not been restored or restore failed - run dotnet restore
2. The project does not list one of 'osx.10.11-x64' in the 'runtimes' section.
3. You may be trying to publish a library. Libraries should be packed with dotnet pack.
```

Or something. Since we can detect the TFM, we might actually say something like "you are publishing netstandard1.5, do not do that, that way be dragons" or something like that.

**Comment by brthor** [comment]
> Or something. Since we can detect the TFM, we might actually say something like "you are publishing netstandard1.5, do not do that, that way be dragons" or something like that.

I think the decision here is whether we want to blacklist specific tfms like `netstandard1.5` or whitelist tfms like `netcoreapp1.0`. There are tradeoffs to each in that blacklisting will give us opportunity to have non-defined errors on new Tfms if they're not blacklisted properly, and whitelisting gives us the opposite opportunity.

I think I prefer the general error message here as I do not know how often we'll be bringing up new Tfms.

**Comment by blackdwarf** [comment]
@brthor agreed, that's why I put both proposals there. The important thing here is for the user to have enough info to start on a solution path. Currently, the error message doesn't help with that because `runtimes` will be the wrong thing for a portable library without native assets.
